HOWTO-make-a-release.md

```diff
@@ -0,0 +1,348 @@
+    -   [Generate the tarball and announcement text](#generating-the-tarball-and-announcement-text) [do not push]
+        -   [OpenSSL 3.0 and on](#openssl-3.0-and-on)
+        -   [OpenSSL before 3.0](#openssl-before-3.0)
+    -   [Update the website locally](#update-the-website-locally) [do not push]
+-   [Publish the release](#publish-the-release)
+-   [Post-publishing tasks](#post-publishing-tasks)
+    -   [Check the website](#check-the-website)
+    -   [Send the announcement mail](#send-the-announcement-mail)
+    -   [Unfreeze the source repository](#unfreeze-the-source-repository)
+    -   [Security fixes](#security-fixes)
+    -   [Keep in touch](#keep-in-touch)
+
+
+# Prerequisites
+
+## Check your access
+
+To perform a release, you must have appropriate access to OpenSSL's
+development host, dev.openssl.org.  To test this, try to log in with
+ssh:
+
+    ssh dev.openssl.org
+
+You must also check that you can perform tasks as the user 'openssl'
+on dev.openssl.org.  When you have successfully logged in, test your
+access to that user with sudo:
+
+    sudo -u openssl id
+
+## PGP / GnuPG key
+
+You must have a PGP / GnuPG key, and its fingerprint should be present
+in the file `doc/fingerprints.txt` in the OpenSSL source.
+
+## Repositories
+
+You must have access to the following repositories:
+
+-   `openssl-git@git.openssl.org:openssl.git`
+
+    This is the usual main source repository
+
+-   `openssl-git@git.openssl.org:openssl-web.git`
+
+    This is the website repository
+
+-   `openssl-git@git.openssl.org:tools.git`
+
+    This contains certain common tools
+
+## A method for reviewing
+
+For reviewing to take place, the release person and the reviewer need a
+way to share changes that are being applied.  Most commonly, that's done
+as PRs through Github.  However, if undisclosed security fixes are being
+applied, some other more private repository must be used.
+Worst case scenario, there have been times where the review process has
+been conducted by sending patches (made with `git format-patch`) by email
+and receiving approval as a reply.
+
+The release person and the reviewer must have a conversation to confirm
+or figure out how review shall be done.
+
+# Pre-publishing tasks
+
+All the actions in this section need to be repeated for each OpenSSL
+version released.
+
+## Freeze the source repository
+
+The day before the release, freeze the main repository.  This locks
+out everyone but the named user, who is doing the release, from doing
+any pushes.  Someone other than the person doing the release should
+run the command.  For example:
+
+    ssh openssl-git@git.openssl.org freeze openssl NAME
+
+## Prepare your repository checkouts
+
+You will need to checkout at least three working trees:
+
+-   one for the website
+
+        git clone openssl-git@git.openssl.org:openssl-web.git website
+
+-   one for extra tools
+
+        git clone openssl-git@git.openssl.org:openssl-tools.git tools
```

Comment:
As well as these tools you also need to have a working version of openssl in your path. Typically this will be on your standard $PATH - but I mention it because its tripped me up several times. I often end up with strange (old or dev) versions of OpenSSL on my $PATH

Reply:
Ah, ok.  I assume that's to calculate the checksums.

Anything else that you can think of?

(side note: who would anyone *not* have openssl in their `$PATH`?????  :wink:)

Reply:
You need gpg too

Reply:
... and ssh...  and git

Reply:
Added